src/commands/watch.ts

```diff
@@ -0,0 +1,181 @@
+        } catch (e) {
+            this.error(e && e.message ? e.message : e, {exit: 2});
+        }
+
+        if (flags.i2fa) {
+            flags.code = await cli.prompt('2FA code', { type: 'hide' });
+        }
+
+        const watcher = chokidar.watch(fd.folder, {
+            ignored: [
+                '**/README.md',
+                '**/package-lock.json',
+                '**/package.json',
+                '**/tslint.json',
+                '**/tsconfig.json',
+                '**/serverInfo.json',
+                '**/*.js',
+                '**/*.js.map',
+                '**/*.d.ts',
+                '**/*.spec.ts',
+                '**/*.test.ts',
+                '**/dist/**',
+                '**/.*',
+            ],
+            awaitWriteFinish: true,
+            persistent: true});
+        if (flags.addfiles) {
+            watcher.add(flags.addfiles);
+        }
+        if (flags.remfiles) {
+            watcher.unwatch(flags.remfiles);
+        }
+        let serverInfo: IServerInfo1 | IServerInfo2;
+        watcher
+            .on('change', async () => {
+                const tasks = new Listr([
+                    {
+                        title: 'Checking report',
+                        task: () => {
+                            try {
+                                checkReport(this, fd, flags);
+                                return;
+                            } catch (e) {
+                                throw new Error(e);
+                            }
+                        },
+                    },
+                    {
+                        title: 'Packaging',
+                        task: async (ctx, task) => {
+                            try {
+                                ctx.zipName = await packageAndZip(this, fd);
+                                return;
+                            } catch (e) {
+                                throw new Error(e);
+                            }
+                        },
+                    },
+                    {
+                        title: 'Reading server info',
+                        task: async (ctx, task)  => {
+                            try {
+                                serverInfo = await getServerInfo(fd);
+                            } catch (e) {
+                                throw new Error(e);
+                            }
+                        },
+                    },
+                    {
+                        title: 'Updating',
+                        task: async (ctx, task) => {
+                            try {
+                                await uploadApp({...flags, update: true, ...serverInfo}, fd, ctx.zipName);
+                            } catch (e) {
+                                throw new Error(e.message);
+                            }
+                        },
+                    },
+                ]);
+                tasks.run().catch((e) => {
+                    return;
```

Comment:
Same as above. This catch block catches error from its child but does nothing, which may make developers cannot find the errors when an error was thrown.